sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -168,4 +168,89 @@ private static boolean isValid(String name) {
   private static boolean isValidAndNotEmpty(AttributeKey<?> name) {
     return !name.getKey().isEmpty() && isValid(name.getKey());
   }
+
+  public static Resource.Builder builder() {
+    return new Resource.Builder();
+  }
+
+  public Builder toBuilder() {
+    return builder().putAll(this);
+  }
+
+  public static class Builder {
```

Comment:
We don't use inner classes, can you move to `ResourceBuilder`?

Reply:
Sure, will do.